lib/internal/modules/esm/module_job.js

```diff
@@ -151,7 +167,33 @@ class ModuleJob {
     await this.instantiate();
     const timeout = -1;
     const breakOnSigint = false;
-    await this.module.evaluate(timeout, breakOnSigint);
+    try {
+      await this.module.evaluate(timeout, breakOnSigint);
+    } catch (e) {
+      if (e?.name === 'ReferenceError' &&
+          isCommonJSGlobalLikeNotDefinedError(e.message)) {
+        e.message += ' in ES module scope';
+
+        if (StringPrototypeStartsWith(e.message, 'require ')) {
+          e.message += ', you can use import instead';
```

Comment:
Perhaps - `Use "import" instead or `

Reply:
Is your comment incomplete or? 😅
For reference, the `e.message` at this point is `ReferenceError: require is not defined is ES module scope`.

Reply:
Was working to `ReferenceError: require is not defined in the ES module ${url}` per the previous comment.

Reply:
Oh I see! Note that the second part is printed out only if the module is `.js`, we need to take into account the case where the module is using a different extension or is a `data:` URL.